sdk-java/src/main/java/io/littlehorse/sdk/wfsdk/internal/WorkflowThreadImpl.java

```diff
@@ -444,19 +452,42 @@ public SpawnedThreadImpl spawnThread(ThreadFunc threadFunc, String threadName, M
         return new SpawnedThreadImpl(this, threadName, internalStartedThreadVar);
     }
 
+    public void overrideTaskRetries(TaskNodeOutputImpl node, int retries) {
+        checkIfIsActive();
+        Node.Builder nb = spec.getNodesOrThrow(node.nodeName).toBuilder();
+        if (nb.getNodeCase() != NodeCase.TASK) {
+            throw new IllegalStateException("Impossible to not have task node here");
+        }
+
+        TaskNode.Builder taskBuilder = nb.getTaskBuilder();
+        taskBuilder.setRetries(retries);
+
+        nb.setTask(taskBuilder);
+        spec.putNodes(node.nodeName, nb.build());
+    }
+
     public void addTimeoutToExtEvt(NodeOutputImpl node, int timeoutSeconds) {
         checkIfIsActive();
         Node.Builder n = spec.getNodesOrThrow(node.nodeName).toBuilder();
-        if (n.getNodeCase() != NodeCase.EXTERNAL_EVENT) {
-            throw new RuntimeException("Tried to set timeout on non-ext evt node!");
-        }
 
-        ExternalEventNode.Builder evt = n.getExternalEventBuilder();
-        evt.setTimeoutSeconds(VariableAssignment.newBuilder()
+        VariableAssignment timeoutValue = VariableAssignment.newBuilder()
                 .setLiteralValue(
-                        VariableValue.newBuilder().setInt(timeoutSeconds).setType(VariableType.INT)));
+                        VariableValue.newBuilder().setInt(timeoutSeconds).setType(VariableType.INT))
+                .build();
+
+        if (n.getNodeCase() == NodeCase.TASK) {
+            TaskNode.Builder task = n.getTaskBuilder();
+            task.setTimeoutSeconds(timeoutSeconds);
+            n.setTask(task);
 
-        n.setExternalEvent(evt);
+        } else if (n.getNodeCase() != NodeCase.EXTERNAL_EVENT) {
+
+            ExternalEventNode.Builder evt = n.getExternalEventBuilder();
+            evt.setTimeoutSeconds(timeoutValue);
+            n.setExternalEvent(evt);
+        } else {
+            throw new RuntimeException("Tried to set timeout on non-ext evt node!");
```

Comment:
what do you think about using the full name instead of the abbreviation for non-ext evt?

Reply:
Good suggestion